common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java

```diff
@@ -73,8 +73,7 @@
             "where c.type = 'DRUG' " +
             "and c.subtype in ('BRAND') " +
             "and c.is_selectable = 1 " +
-            "and (upper(c.name) like upper(concat('%',?1,'%')) " +
-            "or upper(c.code) like upper(concat('%',?1,'%'))) " +
+            "and match(c.name, c.code) against(?1 in boolean mode) > 0 " +
```

Comment:
"like" fetches un-relevant results.
Like this call fetches the ingredients of drug "diabetesee" when searched with the keyword "diabetes".
Changed this to match after testing few queries.

Reply:
I can't tell at a glance whether MySQL will optimize this query properly. Once this is in the test env, please check the request latency here (alternatively, you can execute this SQL directly against the test DB to verify the latency). In the other case that I optimized the SQL query alone was taking ~8s.

Reply:
I just searched with tylenol in sequel pro 
"select distinct c.concept_id from criteria c
where c.type = 'DRUG'
and c.subtype in ('BRAND') 
and c.is_selectable = 1 
and match(c.name,c.code) against('tylenol' in boolean mode);" ~ 39.7 ms
"select distinct c.concept_id from criteria c
where c.type = 'DRUG'
and c.subtype in ('BRAND') 
and c.is_selectable = 1 
and (upper(c.name) like upper(concat('%','tylenol','%')) or upper(c.code) like upper(concat('%','tylenol','%')));" ~ 254ms

Reply:
Against test cloud SQL right? Not local? If so, that's pretty slow but probably acceptable if this is not a super common endpoint.

Reply:
This is against the test cloudsql. I would not say its uncommon. Each time the concept search is hit, this method is called to fetch ingredients of any matching drug brands.
Match seems to work better over like or like for few queries i have tested.
Do we have a better and quick option that could speed up things ?

Reply:
What was the runtime for the previous `LIKE` approach?

I think this is probably fine, I can rerun the load tests afterwards to check.

Reply:
previous like took 254 ms, match took 39.7 ms only for tylenol search
like ~ 215ms, match 35.6 ms for "percocet" search

Reply:
Oh - great. Nice improvement then.